src/topp/DecoyDatabase.cpp

```diff
@@ -165,173 +186,234 @@ class TOPPDecoyDatabase :
+    // Create a ProteaseDigestion object for peptide digestion
     ProteaseDigestion digestion;
     String enzyme = getStringOption_("enzyme").trim();
-    if ((input_type == SeqType::protein) && !enzyme.empty())
-    {
-      digestion.setEnzyme(enzyme);
-    }
-    // check if decoy_string is common decoy string (e.g. decoy, rev, ...)
-    String decoy_string_lower = decoy_string;
-    decoy_string_lower.toLower();
-    bool is_common = false;
-    for (const auto& a : DecoyHelper::affixes)
+    if ((input_type == SeqType::protein) && ! enzyme.empty()) { digestion.setEnzyme(enzyme); }
+    
+    
+ // create neighbor peptides for the relevant peptides given in the FASTA from '-in'
+    String neighbor_file = getStringOption_("Neighbor_Search:neighbor_in");
+ // Check if the neighbor file option is provided
+    if (! neighbor_file.empty())
     {
-      if ((decoy_string_lower.hasPrefix(a) && decoy_string_position_prefix) || (decoy_string_lower.hasSuffix(a) && !decoy_string_position_prefix))
+      String out_neighbor = getStringOption_("out_neighbor");
+      FASTAFile neig;
+      neig.writeStart(out_neighbor);
+      double mz_bin_size = getDoubleOption_("Neighbor_Search:mz_bin_size");
+      double min_shared_ion_fraction = getDoubleOption_("Neighbor_Search:min_shared_ion_fraction");
+      double mass_tolerance = getDoubleOption_("Neighbor_Search:mass_tolerance");
+      int missed_cleavages = getIntOption_("Neighbor_Search:missed_cleavages");
+      digestion.setMissedCleavages(missed_cleavages);
+ // Vector to hold all digested peptides from the input files
+      vector<AASequence> all_peptides;
+
+      for (Size i = 0; i < in.size(); ++i)
       {
-        is_common = true;
+        //FASTAContainer<TFI_File> in_entries {in[i]};
+        neig.readStart(in[i]);
+        while (neig.readNext(entry))
+        {
+          // Vector to hold the peptides generated from digestion
+          vector<AASequence> peptides;
+          // Digest the current entry sequence into peptides
+          digestion.digest(AASequence::fromString(entry.sequence), peptides);
+          // Add the generated peptides to the all_peptides vector
+          all_peptides.insert(all_peptides.end(), peptides.begin(), peptides.end());
+        }
       }
-    }
-    // terminate, if decoy_string is not one of the allowed decoy strings (exit code 11)
-    if (!is_common)
-    {
-      if (getFlag_("force"))
+ // Load the neighbor peptides candidates from the neighbor file
+      FASTAFile neighbor_fasta;
+      vector<FASTAFile::FASTAEntry> neighbor_entries;
+      neighbor_fasta.load(neighbor_file, neighbor_entries);
+      vector<AASequence> digested_candidate_peptides;
+      for (auto& entry : neighbor_entries)
       {
-        OPENMS_LOG_WARN << "Force Flag is enabled, decoys with custom decoy string (not in DecoyHelper::affixes) will not be detected.\n";
+        digestion.digest(AASequence::fromString(entry.sequence), digested_candidate_peptides);
```

Comment:
digested_candidate_peptides contains all digested peptides. I have checked the size. It was greater than 1

Reply:
yes, because it was filled with the results from the last iteration.
All others were deleted:
https://github.com/cbielow/OpenMS/blob/develop/src/openms/source/CHEMISTRY/ProteaseDigestion.cpp#L74

Reply:
okay i'll change today